### ISSUE TITLE
0.971: Builtin `sum` expects bool, not int

### ISSUE THREAD

**Issue by tconbeer** [python/mypy#13204]
**Bug Report**

This behavior was new, after upgrading from 0.961 to 0.971.

I am using the built-in `sum()` to add ints returned by a generator function, but mypy insists that `sum()` operates on a sequence of bools, not ints.

**To Reproduce**

In `file_counter.py`:

```py
from pathlib import Path

p = Path(".")
file_count = sum((1 for f in p.iterdir() if f.is_file))
print(file_count)
```

```sh
$ mypy file_counter.py
```

**Expected Behavior**
```
Success: no issues found in 1 source file
```

**Actual Behavior**

```
./file_counter.py:4:19: error: Generator has incompatible item type "int"; expected "bool"
```

The same error is raised if I change the generator to a list comprehension, or if my generator or list comprehension returns any other int or float. 

However, these other variations do not raise an error from mypy:

```py
sum((True for f in p.iterdir() if f.is_file))
sum((1 for f in p.iterdir()))
```

**Your Environment**

- Mypy version used: `mypy 0.971 (compiled: yes)`
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.8.10
- Operating system and version: `Ubuntu 20.04.4 LTS (GNU/Linux 5.10.16.3-microsoft-standard-WSL2 x86_64)` (Ubuntu running on Windows 11/WSL2), but same behavior observed in CI runners running Linux and MacOS
